modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/CalciteBasicSecondaryIndexIntegrationTest.java

```diff
@@ -120,10 +121,11 @@ public void testIndexLoopJoin() {
             .check();
     }
 
+    /** */
     @Test
     public void testMergeJoin() {
         assertQuery("" +
-            "SELECT d1.name, d2.name FROM Developer d1, Developer d2 WHERE d1.depId = d2.depId")
+            "SELECT /*+ DISABLE_RULE('CorrelatedNestedLoopJoin') */ d1.name, d2.name FROM Developer d1, Developer d2 WHERE d1.depId = d2.depId")
```

Comment:
why do we need it here? test passed without this change too.

Reply:
because in the middle of the task progress this test fails several times because of changed plan. Since this test should verify only correctness of MergeJoin execution, I think it's OK to disable other algorithms explicitly